src/main/scala/org/broadinstitute/dsde/firecloud/EntityClient.scala

```diff
@@ -355,51 +354,64 @@ class EntityClient (requestContext: RequestContext, modelSchema: ModelSchema)(im
+
+      val probeUpsert = pipeline {
+        Post(FireCloudDirectiveUtils.encodeUri(Rawls.entityPathFromWorkspace(workspaceNamespace, workspaceName) + "/batchUpsert"), List.empty[String])
+      }
+
+      probeUpsert.map {
+        case resp if resp.status == NoContent => op(true)
+        case resp => RequestComplete(resp) // bubble up errors
+      }
     }
-    val acceptableProtocols = Seq("https") //for when we inevitably change our mind and need to support others
 
-    if (!acceptableProtocols.contains(pfbUrl.getProtocol)) {
-      Future.successful(RequestCompleteWithErrorReport(BadRequest, "Invalid URL protocol: must be https only"))
-    } else {
+    // validate presigned URL
+    validatePfbUrl(pfbRequest) { pfbUrl =>
+      // empty-array batchUpsert to Rawls to verify workspace existence and permissions
+      validateUpsertPermissions() { _ =>
+        // generate job UUID
+        val jobId = java.util.UUID.randomUUID()
 
-      val futurePerRequestMessage = for {
-        arrowResponse <- callArrow
-        rawlsJsonEntity <- arrowResponse.status match {
-          case OK => Future.successful(arrowResponse.entity)
-          case BadRequest | Unauthorized | Forbidden | NotFound => Future.failed(new FireCloudExceptionWithErrorReport(ErrorReport(arrowResponse.status, arrowResponse.entity.asString)))
-          case _ => Future.failed(new FireCloudException(arrowResponse.status.value + ": " + arrowResponse.entity.asString))
-        }
-        rawlsResponse <- callRawls(rawlsJsonEntity)
-        result <- rawlsResponse.status match {
-          case NoContent => Future.successful(RequestComplete(NoContent))
-          case Unauthorized | Forbidden | NotFound => Future.failed(new FireCloudExceptionWithErrorReport(FCErrorReport(rawlsResponse)))
-          case _ => Future.failed(new FireCloudException(rawlsResponse.status.value + ": " + rawlsResponse.entity.asString))
-        }
-      } yield (result)
+        // the payload to Arrow needs to include the PFB url, job id, workspace info, and user info
+        val user = WorkbenchUserInfo(userInfo.id, userInfo.userEmail)
+        val arrowPayload = pfbRequest.copy(jobId = Option(jobId.toString),
+          workspace = Option(WorkspaceName(workspaceNamespace, workspaceName)),
+          user = Option(user))
 
-      futurePerRequestMessage recover {
-        case e: FireCloudExceptionWithErrorReport => RequestCompleteWithErrorReport(optAkka2sprayStatus(e.errorReport.statusCode).getOrElse(InternalServerError), e.errorReport.message, e)
-        case e: Throwable => RequestCompleteWithErrorReport(InternalServerError, "Import failed. Details: " + e.getMessage, e)
+        // fire-and-forget call Arrow with UUID, workspace info, and presigned URL
+        val gzipPipeline = addHeader (`Accept-Encoding`(gzip)) ~> sendReceive ~> decode(Gzip)
+        gzipPipeline { Post(avroToRawlsURL, arrowPayload) }
```

Comment:
Since we are no longer calling the Arrow service but instead calling a Cloud Function - this will have to change - the url being referenced here would be `https://us-central1-terra-arrow-dev.cloudfunctions.net/avro-import` (for dev).

Reply:
updated, here and in #broadinstitute/firecloud-develop#1889